pkg/streamingpromql/testdata/ours/functions.test

```diff
@@ -682,3 +682,89 @@ eval range from 0 to 7m step 1m last_over_time(some_metric_count[3m])
   some_metric_count{env="prod", cluster="eu"} _ _ _ 0 1 2 3 4
   some_metric_count{env="prod", cluster="us"} _ _ _ 0 2 4 6 8
   some_metric_count{env="prod", cluster="au"} _ _ _ {{count:5}} {{count:10}} {{count:15}} {{count:20}} {{count:25}}
+
+clear
+
+# Test sort / sort_desc.
+load 1m
+  test_metric{case="float 1"}             0+10x4
+  test_metric{case="float 2"}             0+15x4
+  test_metric{case="float with NaN"}      NaN NaN NaN NaN NaN
+  test_metric{case="float with +Inf"}     +Inf +Inf +Inf +Inf +Inf
+  test_metric{case="float with -Inf"}     -Inf -Inf -Inf -Inf -Inf
+  test_metric{case="histogram 1"}         {{count:0 sum:12}} {{count:0 sum:12}} {{count:0 sum:12}} {{count:0 sum:12}} {{count:0 sum:12}}
+  test_metric{case="histogram 2"}         {{count:20 sum:5}} {{count:20 sum:5}} {{count:20 sum:5}} {{count:20 sum:5}} {{count:20 sum:5}}
+  test_metric{case="histogram with NaN"}  {{count:0 sum:NaN}} {{count:0 sum:NaN}} {{count:0 sum:NaN}} {{count:0 sum:NaN}} {{count:0 sum:NaN}}
+  test_metric{case="histogram with +Inf"} {{count:0 sum:Inf}} {{count:0 sum:Inf}} {{count:0 sum:Inf}} {{count:0 sum:Inf}} {{count:0 sum:Inf}}
+  test_metric{case="histogram with -Inf"} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}}
+
+# Sorting of identical values is not stable, so we exclude those from these test cases and check them below.
```

Comment:
Is there why there's no gauntlet? Otherwise we should add these there.

Reply:
Yep, and also the fact that `sort` / `sort_desc` only apply to instant queries, but the gauntlet only does range queries.